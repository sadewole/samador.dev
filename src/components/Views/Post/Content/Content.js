import React from "react"
import PropTypes from "prop-types"

const Content = ({ title, body, date }) => {
  return (
    <article className="mb-15">
      <header className="mb-10">
        <h1>{title}</h1>
        <time className="italic" dateTime={date}>
          Published on {date}
        </time>
      </header>
      <section>
        <div dangerouslySetInnerHTML={{ __html: body }} />
      </section>
    </article>
  )
}

Content.propTypes = {
  title: PropTypes.string,
  date: PropTypes.string,
  body: PropTypes.string,
}

export default Content
