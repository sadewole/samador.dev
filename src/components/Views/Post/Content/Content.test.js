import React from "react"
import { render, screen } from "@testing-library/react"
import renderer from "react-test-renderer"

import Content from "./Content"

describe("Content", () => {
  const props = {
    title: "test title",
    body: "<p>test</p>",
    date: "test date",
  }

  it("must contain string", () => {
    render(<Content {...props} />)

    expect(screen.getByTestId("publish")).toHaveTextContent("Published on")
  })

  it("render correctly", () => {
    const tree = renderer.create(<Content {...props} />).toJSON()
    expect(tree).toMatchSnapshot()
  })
})
