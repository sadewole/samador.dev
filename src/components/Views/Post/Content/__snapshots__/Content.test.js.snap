// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Content render correctly 1`] = `
<article
  className="mb-15"
>
  <header
    className="mb-10"
  >
    <h1>
      test title
    </h1>
    <time
      className="italic"
      data-testid="publish"
      dateTime="test date"
    >
      Published on 
      test date
    </time>
  </header>
  <section>
    <div
      dangerouslySetInnerHTML={
        Object {
          "__html": "<p>test</p>",
        }
      }
    />
  </section>
</article>
`;
